docs/src/main/docs/guides/12_tracing_mp_guide.adoc

```diff
@@ -0,0 +1,759 @@
+
+[source,java]
+.Update the `GreetingProvider` class; 1) Add @Traced to the `GreetingProvider` class and 2) Remove @Traced from the `getMessage` method:
+----
+@Traced // <1>
+@ApplicationScoped
+public class GreetingProvider {
+...
+
+    String getMessage() { // <2>
+        return message.get();
+    }
+----
+<1> This will enable tracing for all class methods, except for the constructor and methods that are private.
+<2> Remove @Traced for the `getMessage` method.
+
+[source,bash]
+.Build and run the application, then invoke the endpoints and check the response:
+----
+curl http://localhost:8080/greet
+curl -X GET "http://localhost:9411/api/v2/spans?serviceName=helidon-mp-1" -H "accept: application/json"
+[
+...
+  "io.helidon.examples.quickstart.mp.greetingprovider.getmessage", // <1>
+...
+]
+----
+<1> The service has the same set of span names as above, since `getmessage` was the only method called in this bean.
+
+Next, invoke HTTP PUT to change the greeting, which will cause `setMessage` to be called.
+
+[source,bash]
+Invoke the endpoints and check the response:
+----
+curl -i -X PUT -H "Content-Type: application/json" -d '{"greeting": "Hi"}'  http://localhost:8080/greet/greeting // <1>
+curl -X GET "http://localhost:9411/api/v2/spans?serviceName=helidon-mp-1" -H "accept: application/json"
+...
+[
+  "content-read",
+  "content-write",
+  "get:io.helidon.examples.quickstart.mp.greetresource.getdefaultmessage",
+  "io.helidon.examples.quickstart.mp.greetingprovider.getmessage",
+  "io.helidon.examples.quickstart.mp.greetingprovider.setmessage", // <2>
+  "put:io.helidon.examples.quickstart.mp.greetresource.updategreeting", // <3>
+  "security",
+  "security:atn",
+  "security:atz",
+  "security:response"
+]
+----
+<1> Invoke the endpoint to change the greeting.
+<2> The `GreetingProvider.setmessage` method was traced since you enabled class level tracing.
+<3> The JAX-RS method `GreetResource.updategreeting` was traced automatically by Helidon.
+
+You can refresh the UI view and drill down the trace to see the new spans.
+
+NOTE: Methods invoked directly by your code are not enabled for tracing, even if you explicitly annotate them with @Traced.
+Tracing only works for methods invoked on CDI beans. See the example below.
+
+[source,java]
+.Update the `GreetingProvider` class with the following code:
+----
+@ApplicationScoped
+public class GreetingProvider {
+    private final AtomicReference<String> message = new AtomicReference<>();
+
+    @Inject
+    public GreetingProvider(@ConfigProperty(name = "app.greeting") String message) {
+        this.message.set(message);
+    }
+
+    @Traced // <1>
+    String getMessage() {
+        return getMessage2();
+    }
+
+    @Traced  // <2>
+    String getMessage2() {
+        return message.get();
+    }
+
+    void setMessage(String message) {
+        this.message.set(message);
+    }
+}
+----
+<1> The `getMessage` method will be traced since it is externally invoked by `GreetingResource`.
```

Comment:
I think it is `GreetResource` not `GreetingResource`

Reply:
fixed